test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
+      XDG_CONFIG_DIRS='siamese:persian',
+      XDG_DATA_HOME='dog',
+      XDG_DATA_DIRS='chihuahua:terrier',
+      XDG_CACHE_HOME='capybara',
+    }})
+  end)
+
+  it('knows the config home', function()
+    eq('cat/nvim', eval('stdpath("config")'))
+  end)
+
+  it('knows the data home', function()
+    if os_name() == 'windows' then
+      eq('dog\\nvim-data', eval('stdpath("data")'))
```

Comment:
Questions:

- `alter_slashes()` is a noop for Unix and converts `/` to `\` on Windows?
- `os_path()` does what? Choses arg1 for Linux and arg2 for Windows?
- Where would these go?
- Do I have to create `funcs.stdpath()` myself (and where)? Or is it created on its own?